### DISCUSSION TITLE
[QUESTION] Vite Vue

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [DanielXMoore/Civet#460] [Q&A] [answered]
How can I integrate this with Vite Vue for usage in .vue files, in the `<script setup lang="ts"> `?

I loved Civet's official docs, I had fun! Now I wish that I can integrate this with Vue and will start promoting it heavily!!!

Thank you for making this dreamy language taking TypeScript to the next level!

Wish you health, peace, and success.

**Comment by angelhdzmultimedia** [comment]
Oh didn't notice there's a Discussions tab. Can you convert this to a Question, please? :)

**Comment by edemaine** [comment, marked as answer]
I believe this issue is tracked in https://github.com/lorefnon/vite-plugin-civet/issues/7
It's not possible currently, but hopefully we will get there. See the discussion about what it would take.

**Reply by angelhdzmultimedia** [reply]
> I believe this issue is tracked in [lorefnon/vite-plugin-civet#7](https://github.com/lorefnon/vite-plugin-civet/issues/7) It's not possible currently, but hopefully we will get there. See the discussion about what it would take.

Thank you very much for the response! Have a nice rest of the day.

**Comment by EstellaDepperschmidtzD** [comment]
integrating Civet with Vue should be relatively straightforward. You can simply import the Civet module and use it within your Vue component like you would with any other module. You can also take advantage of TypeScript support in Vue by using the <script setup lang="ts"> tag.
